tests/subsys/net/lib/download_client/src/main.c

```diff
@@ -95,128 +81,222 @@ static void dl_coap_start(struct download_client *client)
+	test_stubs_teardown();
+}
+
+ZTEST_SUITE(download_client, NULL, NULL, my_suite_before, NULL, NULL);
+
+ZTEST(download_client, test_download_ok)
 {
 	struct download_client client;
-	int32_t recvfrom_params[] = { 25, 25, 25 };
-	int32_t sendto_params[] = { 20, 20, 20 };
+	size_t socket_recvfrom_ret_values[] = { 25, 25, 25 };
+	size_t socket_sendto_ret_values[] = { 20, 20, 20 };
 
-	dl_coap_init(75, 20);
+	coap_get_recv_timeout_fake.custom_fake = coap_get_recv_timeout_fake_default;
+	coap_parse_fake.custom_fake = coap_parse_fake_default;
+	coap_request_send_fake.custom_fake = coap_request_send_fake_default;
 
-	mock_return_values("mock_socket_offload_recvfrom", recvfrom_params,
-			   ARRAY_SIZE(recvfrom_params));
-	mock_return_values("mock_socket_offload_sendto", sendto_params, ARRAY_SIZE(sendto_params));
+	test_set_ret_val_socket_offload_recvfrom(socket_recvfrom_ret_values,
+						 ARRAY_SIZE(socket_recvfrom_ret_values));
+	test_set_ret_val_socket_offload_sendto(socket_sendto_ret_values,
+					       ARRAY_SIZE(socket_sendto_ret_values));
 
-	dl_coap_start(&client);
+	download_start(&client);
 
-	zassert_ok(wait_for_event(DOWNLOAD_CLIENT_EVT_DONE, 10), "Download must have finished");
+	zassert_true(wait_for_event(DOWNLOAD_CLIENT_EVT_DONE), "Download must have finished");
+	zassert_equal(coap_get_recv_timeout_fake.call_count, 3, "");
+	zassert_equal(coap_request_send_fake.call_count, 3, "");
+	zassert_equal(coap_parse_fake.call_count, 3, "");
 
-	de_init(&client);
+	download_stop(&client);
 }
 
-static void test_download_reconnect_on_socket_error(void)
+ZTEST(download_client, test_download_reconnect_on_socket_error_ok)
 {
 	struct download_client client;
-	int32_t recvfrom_params[] = { 25, -1, 25, 25 };
-	int32_t sendto_params[] = { 20, 20, 20, 20 };
+	size_t socket_recvfrom_ret_values[] = { 25, -1, 25, 25 };
+	size_t socket_sendto_ret_values[] = { 20, 20, 20, 20 };
 
-	dl_coap_init(75, 20);
+	coap_get_recv_timeout_fake.custom_fake = coap_get_recv_timeout_fake_default;
+	coap_parse_fake.custom_fake = coap_parse_fake_default;
+	coap_request_send_fake.custom_fake = coap_request_send_fake_default;
 
-	mock_return_values("mock_socket_offload_recvfrom", recvfrom_params,
-			   ARRAY_SIZE(recvfrom_params));
-	mock_return_values("mock_socket_offload_sendto", sendto_params, ARRAY_SIZE(sendto_params));
+	test_set_ret_val_socket_offload_recvfrom(socket_recvfrom_ret_values,
+						 ARRAY_SIZE(socket_recvfrom_ret_values));
+	test_set_ret_val_socket_offload_sendto(socket_sendto_ret_values,
+					       ARRAY_SIZE(socket_sendto_ret_values));
 
-	dl_coap_start(&client);
+	download_start(&client);
 
-	zassert_ok(wait_for_event(DOWNLOAD_CLIENT_EVT_DONE, 10), "Download must have finished");
+	zassert_true(wait_for_event(DOWNLOAD_CLIENT_EVT_DONE), "Download must have finished");
+	zassert_equal(coap_get_recv_timeout_fake.call_count, 4, "");
+	zassert_equal(coap_request_send_fake.call_count, 4, "");
+	zassert_equal(coap_parse_fake.call_count, 3, "");
```

Comment:
You don't need to pass `""` to these macros, simply omit the parameter in the call.